front/components/assistant/conversation/AgentMessage.tsx

```diff
@@ -365,6 +369,7 @@ export function AgentMessage({
   }
 
   async function retryHandler(agentMessage: AgentMessageType) {
+    setIsRetryHandlerProcessing(true);
```

Comment:
nit: I usually separate the handler logic from the UX logic (such as setting a disabled state) that I put in the onClick
feel free to discard

Reply:
I don't have a strong opinion on this but usually I'd rather group the state update and api call, to avoid reusing the function elsewhere and not having the state updated.

I will discard and keep as is, especially as `retryHandled` is used multiple times in the file and I don't want to duplicated the state logic 🙏🏻